src/renderer/screens/Editor/Macros/MacroStep.js

```diff
@@ -0,0 +1,88 @@
+// -*- mode: js-jsx -*-
+/* Chrysalis -- Kaleidoscope Command Center
+ * Copyright (C) 2020-2022  Keyboardio, Inc.
+ *
+ * This program is free software: you can redistribute it and/or modify it under
+ * the terms of the GNU General Public License as published by the Free Software
+ * Foundation, version 3.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+import React from "react";
+import { useTranslation } from "react-i18next";
+
+import Chip from "@mui/material/Chip";
+import DeleteIcon from "@mui/icons-material/Delete";
+
+import { KeymapDB } from "@api/keymap";
+
+const db = new KeymapDB();
+
+const MacroStep = (props) => {
+  const { step, index, selected } = props;
+  const { t } = useTranslation();
+
+  const onDelete = () => {
+    props.onDelete(index);
+  };
+
+  const onClick = () => {
+    props.onClick(index);
+  };
+
+  const formatStep = (step) => {
+    if (step.type == "INTERVAL" || step.type == "WAIT")
```

Comment:
Is there a reason we want macro step types to be strings rather than constants we can compare against?

Reply:
The only reason is that we also use the strings for translations, like: `return t("editor.macros.steps." + step.type);` a few lines below.

We could, of course, have a function to map constants to strings or something along those lines, but that feels more complicated. More correct, but also more complicated. Is the added complexity worth it?

Reply:
The types could be string constants, which really just means we'd need to define them somewhere. This mostly feels like a "not hurting ourselves down the road" thing.

Reply:
> [61359c0](https://github.com/keyboardio/Chrysalis/commit/61359c0f56cff3aa280701610b3f8fde9d29bf90) looks like a nice improvement, but I'm not entirely sure I understand the "why" of the changes from the commit message.

ACK. I'll update the commit message.

The goal is to let us use the sidebar both for changing both keys on the keymap, and the keys used during a macro, with the same sidebar. We're passing `props.onKeyChange` to the Sidebar already, so this PR modifies `onKeyChange`: if the macro editor is open, key change is handled one way, if it isn't, it's handled another way. For this to work, I needed to stop the Sidebar from doing layer + index-based lookups instead (as that doesn't make sense for macro steps), and accept a `props.currentKey` property instead, to let the Editor control what the Sidebar sees, rather than both doing the lookups themselves.